### DISCUSSION TITLE
Active Window Icon in i3bar

### DISCUSSION THREAD

**Discussion by pavanbhat1999** [i3/i3#4791] [Q&A] [answered]
After the release of 4.20 there is support to show window title bar with icons.

I was curious is there a way to add this title bar along with icon to the i3bar?

Reason I want this feature is to utilise the space in i3bar effectively. Usually I work on 2 or 3 workspaces active remaining space in bar is empty.

It would be nice to see current active window with icon. I know that you can display window title as text but I also would like to have icon with it.

Any scripts or suggestions are welcome 🤗🤗.

Note : If anyone familiar with awesome window manager It has this kind of feature built into it.

![screenshot-20220114-14:25:35](https://user-images.githubusercontent.com/47946571/149487235-371507f1-a060-4c9e-b952-ec106ab52a31.png)

This is a fork of i3 which i am using where it is possible. Will this kind of feature be merged

**Comment by Airblader** [comment, marked as answer]
The philosophy of i3bar is to show workspaces, not windows, so there is no configuration way of doing it. You'd have to write a script that shows the icon in the status bar, renames the workspace to that icon or something like that.

**Comment by pavanbhat1999** [comment]
I dont want to rename the workspace . I just want to stick default window title into the i3bar.
Is there a way I  can i retrive title_window_jcon using some command , If thats possible then I can easily write a script.
 
 Is it possible in any way

**Reply by Airblader** [reply]
> Is there a way I can i retrive title_window_jcon using some command

No. You need to read the icon from the client window itself, as it has been possible before i3 4.20 already anyway.

**Comment by pavanbhat1999** [comment]
Okay lets see 
I will use the fork for now.
